### PR TITLE
Update array_helpers.py

### PR DESCRIPTION
**Pull request by moriisaac**
array_helpers.py/Fix sums_products_differences.test_numpy_nancumprod #9886 #9908

### CODE REVIEW

**Review by hirwa-nshuti — COMMENTED** [review]
Hey @moriisaac 

Given that @CatB1t is the one familiar with these helper functions I am requesting a review from him once he approved the changes we will merge.
Thanks😊

**Review by CatB1t — COMMENTED** [review]
Hello @moriisaac , can you provide more context/description of what is the purpose of the PR? thanks!

**Comment by moriisaac** [comment]
Hey @hirwa-nshuti  and @CatB1t , 
the  'numpy.random.randn()' is a function that allows you to generate random float values, including subnormal values.
The PR is supposed to handle the case where the floats() function in array_helpers.py doesn't have a "allow_subnormal" keyword argument, it will print a message that the function doesn't support the argument and it will continue to use the floats function.
This solves the issue #9886Fix sums_products_differences.test.numpy_nancumprod

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 1]
```diff
@@ -780,6 +780,22 @@ def array_values(
                         exclude_max=exclude_max,
                     ),
                 )
+                if "float" in dtype or "complex" in dtype:
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:783]: I'm not quite sure why we're doing these checks once again. They're done earlier in the function

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py [hunk 2]
```diff
@@ -780,6 +780,22 @@ def array_values(
...
+                    kind_dtype = "float"
+                    dtype_info = ivy.finfo(dtype)
+                elif "int" in dtype:
+                    kind_dtype = "int"
+                    dtype_info = ivy.iinfo(dtype)
+                elif "bool" in dtype:
+                    # new code to generate random float array with subnormal values
+                    if allow_subnormal:
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:791]: This logic doesn't seem intuitive to me, can you explain further why is this done in `array_helpers`? we shouldn't be generally assuming that kind of input to produce a specific case for the function we're trying to fix, an input of type `bool` should **ALWAYS** produce an array of booleans and ignore any other arguments related to other data types. 

I'd suggest solving that specific case by using a composite strategy instead.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py (hunk 1 above)
Comment (moriisaac) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:783]: I implented the checks cause on running the function nancumprod, it rendered the error, floats function does not allow the 'allow_subnormal' argument that's why i Had to implement the checks

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py (hunk 2 above)
Comment (moriisaac) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:791]: Okay, I get, if you go through my comments I have explaimed why I  had it in array_helpers. Yes, I understand the concern about assuming that a specific data type should produce a specific case for the function. It's always better to have a more flexible and composite strategy that takes into account multiple scenarios and inputs. Let me correct that

**Comment by nathzi1505** [comment]
Hey @moriisaac , apologies for the delay.
Thank you so much for the commits :D.
I see that there are some lint issues.
Would you mind reviewing them and asking for a recheck again?

**Comment by moriisaac** [comment]
Hey @nathzi1505  Thank you for your reply let me get through with the checks

**Comment by moriisaac** [comment]
Hey @nathzi1505  I resolved all the lint errors the code is good to go😎

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by moriisaac** [comment]
Is it merged?

**Comment by VedPatwardhan** [comment]
Hey @moriisaac, actually the PR got closed automatically as we did a `git push --force` on master this weekend, could you please create a new PR with your changes? Feel free to check out the announcement on the discord server to understand what was done and how to fix the issues that'll arise on your branch to get it up to speed with master. Thanks 🙂

**Pull request closed without merging**